### PR TITLE
Feature/Byte Vec Deserialize Wrapper

### PR DESCRIPTION
**Pull request by unizippro**
Wrapper for deserializing a Vector of bytes like fx "+UUWLE:32AB6A7A4044".
Primarily done to parse bssid for [ublox-short-range-rs](https://github.com/BlackbirdHQ/ublox-short-range-rs/).
But this is kind of a catch all for any Non comma separated value in any AT command.

### CODE REVIEW

File: serde_at/src/de/mod.rs [hunk 1]
```diff
@@ -737,8 +806,23 @@ mod tests {
...
     }
+
+    #[test]
+    fn char_vec_struct() {
+        #[derive(Debug, Deserialize, PartialEq)]
+        struct CharVecTest(CharVec<consts::U4>);
+        let stru = CharVecTest(CharVec(
+            heapless::Vec::from_slice(&['I', 'M', 'P', '_']).unwrap(),
+        ));
+        assert_eq!(crate::from_str("+CCID: IMP_"), Ok(stru));
```
Comment (MathiasKoch) [inline serde_at/src/de/mod.rs:826]: No reason for the extra wrapper here?

```suggestion
        let expectation = CharVec(
            heapless::Vec::from_slice(&['I', 'M', 'P', '_']).unwrap(),
        );
        assert_eq!(crate::from_str("+CCID: IMP_"), Ok(expectation));
```

**Review by MathiasKoch — CHANGES_REQUESTED** [review]
A few small changes, but overall looks good.

I realize it is not quite the same case, 
but perhaps you could "un-ignore" the existing test-case for `Vec` and make that pass as well?

File: serde_at/src/de/seq.rs [hunk 2]
```diff
@@ -42,7 +42,6 @@ impl<'a, 'de> de::SeqAccess<'de> for SeqAccess<'a, 'de> {
     }
 }
 
-#[allow(clippy::module_name_repetitions)]
 pub struct SeqByteAccess<'a, 'b> {
```
Comment (MathiasKoch) [inline serde_at/src/de/seq.rs:45]: Okay, i stand corrected on the `#[allow(clippy::module_name_repetitions)]`

File: serde_at/src/de/mod.rs [hunk 3]
```diff
@@ -26,13 +26,21 @@ pub type Result<T> = core::result::Result<T, Error>;
...
 /// #[derive(Debug, Deserialize, PartialEq)]
-/// struct CharVecStruct(CharVec<consts::U7>)
+/// struct CommandStruct{
+///     id: u8,
+///     vec: CharVec<consts::U7>
+///     value: i32,
+/// }
 ///
-/// let vec: CharVecStruct = from_str("+CCID: IMP_MSG")
+/// let incomming: CommandStruct = from_str("+CCID: 4,IMP_MSG,-12")
```
Comment (MathiasKoch) [inline serde_at/src/de/mod.rs:35]:
```suggestion
/// let incoming: CommandStruct = from_str("+CCID: 4,IMP_MSG,-12")
```

File: serde_at/src/de/mod.rs [hunk 4]
```diff
@@ -26,13 +26,21 @@ pub type Result<T> = core::result::Result<T, Error>;
...
 ///
-/// let stru = CharVecTest(CharVec(heapless::Vec::from_slice(&['I', 'M', 'P', '_', 'M', 'S', 'G']).unwrap()));
+/// let expected = CommandStruct{ 
+///     id: 4,
+///     vec : CharVec(heapless::Vec::from_slice(&['I', 'M', 'P', '_', 'M', 'S', 'G']).unwrap()),
+///     value: -12,
+/// }
 ///
-/// assert_eq!(vec, Ok(stru));
+/// assert_eq!(incomming, Ok(expected));
```
Comment (MathiasKoch) [inline serde_at/src/de/mod.rs:43]:
```suggestion
/// assert_eq!(incoming, Ok(expected));
```

**Comment by unizippro** [comment]
Tarpaulin is crashing and i have no clue why this is the case.
Anyone who can offer any advice or help would be greatly appreciated!

**Review by MathiasKoch — APPROVED** [review]

**Pull request merged**